### PR TITLE
Add artifact signature information to the generated index

### LINKED ISSUE #306: Add signatures to the generated index
**Motivation**

When we release plugins we can now add signatures and check them (see example: https://github.com/falcosecurity/falcoctl/pull/305)
Now we need to make sure that the index file we generate contains the signature section accordingly

Note that for plugins, the index file could in the future also contain plugins that are not generated from this repository, so we may want to rely to those third-party trusted plugin authors to supply that signature and add it to this repo.

### PR DESCRIPTION
**Pull request by maxgio92**
**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area registry
/area build

**What this PR does / why we need it**:

This PR is needed to provide required information for consumers in order to verify OCI signatures of plugins officially distributed as OCI artifacts.

The `registry.yaml` index is consumed by current CI to generate the general Falco artifacts index https://github.com/falcosecurity/falcoctl/blob/gh-pages/index.yaml, which can be further consumed by [falcoctl](https://github.com/falcosecurity/falcoctl) to verify artifacts (i.e. plugins) signatures.

**Which issue(s) this PR fixes**:

Fixes #306 

**Special notes for your reviewer**:

Only one signin implementation is currently supported, which is based on cosign. When signatures are generated by cosign in keyless mode, can provide the OIDC issuer and certificate identity (see https://github.com/falcosecurity/falcoctl/pull/305).
 
Furthermore, this PR adds black box and white box unit tests.

The same feature should be applied to the registry tool of the [rules](https://github.com/falcosecurity/rules).

### CODE REVIEW

File: build/registry/pkg/distribution/index_internal_test.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	"reflect"
+	"testing"
+
+	"github.com/falcosecurity/falcoctl/pkg/index"
+
+	"github.com/falcosecurity/plugins/build/registry/pkg/distribution"
+	"github.com/falcosecurity/plugins/build/registry/pkg/registry"
+)
+
+func TestPluginToIndexEntrySignature(t *testing.T) {
```
Comment (maxgio92) [inline build/registry/pkg/distribution/index_internal_test.go:29]: These unit tests assert the minimum expectations from this feature.

File: build/registry/pkg/distribution/index_test.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+const (
+	indexFile         = "testdata/index.yaml"
+	wrongIndexFile    = "testdata/wrong-index.yaml"
+	registryFile      = "testdata/registry.yaml"
+	wrongRegistryFile = "testdata/wrong-registry.yaml"
+	registryUser      = "falcosecurity"
+	registryName      = "ghcr.io"
+)
+
+var _ = Describe("Update index", func() {
```
Comment (maxgio92) [inline build/registry/pkg/distribution/index_test.go:37]: These black box unit test provide the minimum expected behaviour with specifications of the index update from the registry manifest.

**Comment by leogr** [comment]
Hey @maxgio92 

I lost track of this. What's the status? Is this ready to be merged? :thinking: 

cc @jasondellaluce @LucaGuerra 

PS I see some tests failing.

**Comment by maxgio92** [comment]
Hi @leogr, sorry for the delay. The PR is ready to be reviewed.

File: build/registry/go.mod [hunk 3]
```diff
@@ -3,120 +3,260 @@ module github.com/falcosecurity/plugins/build/registry
 go 1.19
 
 require (
-	github.com/aws/aws-sdk-go-v2 v1.16.16
+	github.com/aws/aws-sdk-go-v2 v1.18.1
 	github.com/aws/aws-sdk-go-v2/feature/s3/manager v1.11.34
 	github.com/aws/aws-sdk-go-v2/service/s3 v1.27.11
 	github.com/blang/semver v3.5.1+incompatible
-	github.com/falcosecurity/falcoctl v0.3.0-rc6
+	github.com/falcosecurity/falcoctl v0.5.2-0.20230707100440-5e6ce83dedba
```
[the remark this answers is not in the document]
Comment (LucaGuerra) [reply build/registry/go.mod:10]: We have `0.6.1` released now :)

**Comment by maxgio92** [comment]
Hey @LucaGuerra, I've just rebased and upgraded falcoctl to v0.6.1

**Comment by LucaGuerra** [comment]
Thank you Max!

I have performed a complete test of this patch by:
1. signing all the artifacts with a separate PR
2. updating an index file with this code
3. using `falcoctl` to install all artifacts and testing signatures

and it worked :tada: 

I think we can merge this :rocket:

**Review by leogr — APPROVED** [review]

**Pull request merged**